### PR TITLE
RPC call timeout implementation

### PR DESCRIPTION
**Pull request by j4r0u53k**

### CODE REVIEW

**Comment by j4r0u53k** [comment]
@fvacek @syyyr

File: src/serialrw.rs [hunk 1]
```diff
@@ -84,76 +96,202 @@ impl<R: AsyncRead + Unpin + Send> SerialFrameReader<R> {
...
+                }
+                if self.data.len() > 1 {
+                    let protocol = self.data[0];
+                    if protocol != Protocol::ChainPack as u8 {
+                        log!(target: "Serial", Level::Debug, "Not chainpack message");
+                        continue 'read_frame
+                    }
+                    let mut buffrd = BufReader::new(&self.data[1 ..]);
+                    let mut rd = ChainPackReader::new(&mut buffrd);
+                    if let Ok(Some(meta)) = rd.try_read_meta() {
```
Comment (fvacek) [inline src/serialrw.rs:140]: It seems that try_read_meta() is called on every single character received until valid meta data is parsed

Comment (j4r0u53k) [reply src/serialrw.rs:140]: Yes, that's true, not very optimal way, but there is no control character between metadata and data. Any suggestions to do it better?

**Comment by fvacek** [comment]
Implemented in https://github.com/silicon-heaven/libshvrpc-rs/pull/11

**Pull request closed without merging**